src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -96,9 +99,13 @@ private void processZipFiles(CloudBlobContainer container)
 
         ServiceBusHelper serviceBusHelper = serviceBusHelper();
 
-        for (ListBlobItem blobItem : container.listBlobs()) {
-            String zipFilename = FilenameUtils.getName(blobItem.getUri().toString());
-
+        // Randomise iteration order to minimise lease acquire contention
+        List<String> zipFilenames = new ArrayList<>();
+        container.listBlobs().forEach(
+            b -> zipFilenames.add(FilenameUtils.getName(b.getUri().toString()))
+        );
+        Collections.shuffle(zipFilenames);
+        for (String zipFilename: zipFilenames) {
```

Comment:
Apparently there's a more efficient way of randomly accessing list elements. Which is more/less extraction of `Collections.shuffle`:

```java
Random random = new Random();
int listSize = list.size();

for (int i = 0; i < listSize; i++) {
    int index = rand.nextInt(listSize));
    String element = list.get(index);
    // do stuff
}
```

Reply:
Yes it seems right as I suppose this would avoid the penalty of rewriting the list (which is what shuffle does) before iterating. I am not sure in our case this would bring about much of a gain though.